src/code/z_camera.c

```diff
@@ -1,3070 +1,6778 @@
+    Vec3f* at = &camera->at;
+    Vec3f* eye = &camera->eye;
+    Vec3f* eyeNext = &camera->eyeNext;
+
+    Vec3f eyePos;
+    s32 atEyeBgId;
+    s32 eyeAtBgId;
+    s32 ret;
+    f32 cosEyeAt;
 
-s32 func_800457A8(Camera* camera, VecSph* arg1, f32 arg2, s16 arg3) {
-    f32 unused;
-    Vec3f sp50;
-    Vec3f sp44;
-    s32 unused2;
-    PosRot* sp2C;
-    f32 temp_ret;
+    eyeChk->pos = camera->eyeNext;
 
-    temp_ret = Player_GetCameraYOffset(camera->player);
+    ret = 0;
 
-    sp50.x = 0.f;
-    sp50.z = 0.f;
-    sp50.y = temp_ret + arg2;
+    atEyeBgId = Camera_BGCheckInfo(camera, at, eyeChk);
+    if (atEyeBgId != 0) {
+        // collision found between at->eye
+        atChk->pos = camera->at;
 
-    sp2C = &camera->playerPosRot;
-    if (arg3 != 0) {
-        sp50.y -= OLib_ClampMaxDist(func_80045714(&camera->unk_108, sp2C->rot.y, arg1->yaw, OREG(9)), temp_ret);
+        OLib_Vec3fToVecSphGeo(&eyeChk->sphNorm, &eyeChk->norm);
+
+        if (eyeChk->sphNorm.pitch >= 0x2EE1) {
+            eyeChk->sphNorm.yaw = diffSph->yaw;
+        }
+
+        eyeAtBgId = Camera_BGCheckInfo(camera, eyeNext, atChk);
+
+        if (eyeAtBgId == 0) {
+            // no collision from eyeNext->at
+            if (checkEye & 1) {
+
+                atChk->pos = *at;
+                eyePos = *eye;
+
+                if (Camera_BGCheckInfo(camera, &eyePos, atChk) == 0) {
+                    // no collision from eye->at
+                    return 3;
+                } else if (eyeChk->poly == atChk->poly) {
+                    // at->eye and eye->at is the same poly
+                    return 3;
+                }
+            } else {
+                return 3;
+            }
+        } else if (eyeChk->poly == atChk->poly) {
+            // at->eyeNext and eyeNext->at is the same poly
+            return 3;
+        }
+
+        OLib_Vec3fToVecSphGeo(&atChk->sphNorm, &atChk->norm);
+
+        if (atChk->sphNorm.pitch >= 0x2EE1) {
+            atChk->sphNorm.yaw = BINANG_ROT180(diffSph->yaw);
+        }
+
+        if (atEyeBgId != eyeAtBgId) {
+            // different bgIds for at->eye[Next] and eye[Next]->at
+            ret = 3;
+        } else {
+            cosEyeAt = Math3D_Cos(&eyeChk->norm, &atChk->norm);
+            if (cosEyeAt < -0.5f) {
+                ret = 6;
+            } else if (cosEyeAt > 0.5f) {
+                ret = 3;
+            } else {
+                ret = 2;
+            }
+        }
     }
-    func_80043A3C(&sp50, &camera->unk_E4, camera->unk_CC.y, camera->unk_CC.x, 0.1f);
+    return ret;
+}
 
-    sp44.x = sp2C->pos.x + camera->unk_E4.x;
-    sp44.y = sp2C->pos.y + camera->unk_E4.y;
-    sp44.z = sp2C->pos.z + camera->unk_E4.z;
-    func_80043A3C(&sp44, &camera->at, camera->unk_100, camera->unk_100, 0.2f);
+/**
+ * Calculates the a value which to adjust the y value when on a slope
+ */
+f32 Camera_CalcSlopeYAdj(Vec3f* floorNorm, s16 playerYRot, s16 eyeAtYaw, f32 adjAmt) {
```

Comment:
this is just a typo and `a` is not a value name right?
can maybe be phrased a bit clearer like 
"Calculate Y adjustment value when on slopes"
or something like that. possibly include a bit more context like the Y for what? position? 
if its angle could possibly use `yaw` a bit more to make the separation (like in the playerYRot name)